src/datastructure/rollback_unionfind.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include <bits/stdc++.h>
+using namespace std;
+
+struct RollbackUnionFind {
+    vector<int> data;
+    stack<pair<int, int>> history;
+    int inner_snap = 0;
+    RollbackUnionFind(int n) { data.resize(n, -1); }
+
```

Comment:
```suggestion

int find(int x) { return data[x] < 0 ? x : find(data[x]); }
```

unite()でfind()を使用するので、先に宣言しておくといいでしょうか？
あと、一応三項演算子verにしましたが、可読性は落ちるのでif文のままでも良いです。